shared/types/form/form.ts

```diff
@@ -104,7 +109,9 @@ export interface FormBase {
 
   responseMode: FormResponseMode
 
-  payments?: FormPayments
+  payments_channel?: FormPaymentsChannel
+
+  payments_field?: FormPaymentsField
```

Comment:
somewhat tangentially related to your PR, I wonder if the payments settings should be moved to `StorageFormBase` instead. Initially when we put this here it was because we were planning to support both email and storage. But since we only plan to support Storage mode for payments, maybe we should move these into their appropriate place. A bunch of other stuff will have to change, including the schema definitions, but I think it's worth it. What do you think?

Reply:
This would also make sense with my other thoughts above

Reply:
Yup I agree, didn't know the consideration for email mode was there before haha

In the same note, should I implement that in this PR, or do it in a separate PR to disable admin form payment settings and field?

Cause currently in email mode, we can still connect stripe and edit the payment drawer, which is undesired. #5819

Reply:
yup no worries! this kind of thing we'll just clean up as we go along haha. I think we should do this now, since you're touching the form schema anyway. 

Re: disabling payment settings and field in email mode - we should do all these checks anyway in fe/be, but moving the `payment_field` and `payment_channel` to `StorageModeBase` will give us model-level validation, so even without implementing fe/be checks, the document save should still fail.

Reply:
Yup okay sounds good! Thanks Justyn will make the adjustments!